src/mbgl/map/map_context.hpp

```diff
@@ -40,13 +45,11 @@ class MapContext : public Style::Observer {
 
     void pause();
 
-    void resize(uint16_t width, uint16_t height, float ratio);
-
     using StillImageCallback = std::function<void(std::exception_ptr, std::unique_ptr<const StillImage>)>;
 
     void triggerUpdate(const TransformState&, Update = Update::Nothing);
-    void renderStill(const TransformState&, StillImageCallback callback);
-    RenderResult renderSync(const TransformState&);
+    void renderStill(const TransformState&, const FrameData&, StillImageCallback callback);
```

Comment:
The TransformState is a status that is set for the map in general. It contains the extent (width/height) and position and is used to determine the tiles we need to load. The Framebuffer size is independent of the logical dimensions of the map and is always associated with a single frame, whereas TransformState represents the overall state of that map and isn't tied to a particular frame.

I plan to move additional data into `FrameData`, such as the `animationTime`.